### PR TITLE
EnvironmentVariableMutatorOptions API proposal

### PR DESCRIPTION
**Pull request by Tyriar**
Part of #179476

### CODE REVIEW

File: src/vs/workbench/api/common/extHostTerminalService.ts [hunk 1]
```diff
@@ -892,15 +892,18 @@ class EnvironmentVariableCollection implements vscode.EnvironmentVariableCollect
...
 	append(variable: string, value: string, scope?: vscode.EnvironmentVariableScope): void {
-		this._setIfDiffers(variable, { value, type: EnvironmentVariableMutatorType.Append, scope });
+		// TODO: Implement options in API
+		this._setIfDiffers(variable, { value, type: EnvironmentVariableMutatorType.Append, scope, options: {} });
 	}
 
 	prepend(variable: string, value: string, scope?: vscode.EnvironmentVariableScope): void {
-		this._setIfDiffers(variable, { value, type: EnvironmentVariableMutatorType.Prepend, scope });
+		// TODO: Implement options in API
+		this._setIfDiffers(variable, { value, type: EnvironmentVariableMutatorType.Prepend, scope, options: {} });
```
Comment (Tyriar) [inline src/vs/workbench/api/common/extHostTerminalService.ts:906]: Currently blocked on a conflict with the old `scopes?` proposal from https://github.com/microsoft/vscode/issues/171173

**Comment by Tyriar** [comment]
When setting `applyInShellIntegration` in the terminal-sample:

image
image

PATH:

image

File: src/vs/workbench/api/browser/mainThreadTerminalService.ts [hunk 2]
```diff
@@ -407,6 +407,7 @@ export class MainThreadTerminalService implements MainThreadTerminalServiceShape
 				map: deserializeEnvironmentVariableCollection(collection),
 				descriptionMap: deserializeEnvironmentDescriptionMap(descriptionMap)
 			};
+			console.log('translatedCollection', translatedCollection);
```
Comment (sandy081) [inline src/vs/workbench/api/browser/mainThreadTerminalService.ts:410]: Is this intended?

**Review by sandy081 — DISMISSED** [review]

File: src/vs/workbench/api/browser/mainThreadTerminalService.ts (hunk 2 above)
Comment (Tyriar) [reply src/vs/workbench/api/browser/mainThreadTerminalService.ts:410]: Nope, thanks for spotting

**Review by sandy081 — DISMISSED** [review]

**Review by sandy081 — DISMISSED** [review]

**Review by karrtikr — DISMISSED** [review]

File: src/vscode-dts/vscode.proposed.envCollectionOptions.d.ts [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+	/**
+	 * Options applied to the mutator.
+	 */
+	export interface EnvironmentVariableMutatorOptions {
+		/**
+		 * Apply to the environment just before the process is created.
+		 *
+		 * Defaults to true.
+		 */
+		applyAtProcessCreation?: boolean;
```
Comment (karrtikr) [inline src/vscode-dts/vscode.proposed.envCollectionOptions.d.ts:19]: It's not following: https://github.com/Microsoft/vscode/wiki/Extension-API-guidelines#undefined-is-false

Comment (Tyriar) [reply src/vscode-dts/vscode.proposed.envCollectionOptions.d.ts:19]: Let's bring this up at the API sync, undefined not being true would be harmful here imo.

**Review by karrtikr — APPROVED** [review]

**Review by karrtikr — APPROVED** [review]

File: src/vs/workbench/api/common/extHostTerminalService.ts [hunk 4]
```diff
@@ -899,19 +905,31 @@ class EnvironmentVariableCollection {
 		return scopedCollection;
 	}
 
-	replace(variable: string, value: string, scope: vscode.EnvironmentVariableScope | undefined): void {
-		this._setIfDiffers(variable, { value, type: EnvironmentVariableMutatorType.Replace, scope });
+	replace(variable: string, value: string, options: vscode.EnvironmentVariableMutatorOptions | undefined, scope: vscode.EnvironmentVariableScope | undefined): void {
+		if (this._extension && options) {
+			isProposedApiEnabled(this._extension, 'envCollectionOptions');
```
Comment (karrtikr) [inline src/vs/workbench/api/common/extHostTerminalService.ts:910]: @Tyriar This seems to return a boolean but isn't consumed anywhere, did we mean to use `checkProposedApiEnabled` instead?

Comment (karrtikr) [reply src/vs/workbench/api/common/extHostTerminalService.ts:910]: I've changed it in https://github.com/microsoft/vscode/pull/183255.

**Pull request merged**